### PR TITLE
Return results in Operation event for visibility of actions

### LINKED ISSUE #193: Emit event for each bundle action performed by BLSWallet
### Problem
When `VerificationGateway.processBundle` is executed, if the encoded functions that are run from actions in `BLSWallet._performOperation` emit any events, it can be hard to determine which events came from which actions.

### Solution
- Emit an event after each action run in `BLSWallet._performOperation` with action metadata including information like action index, value, contract address, etc.

### Considerations

This might not be worth doing if it will add significant gas overhead to `VerificationGateway.processBundle`.

### PR DESCRIPTION
**Pull request by jacque006**
## What is this PR doing?
Making actions and (failing results) more accessible. 

## How can these changes be manually tested?
Unit test for failing action.
Look at Operation event logs of for action data when all succeed, and an error message when one action fails.

## Does this PR resolve or contribute to any issues?
Closes #193 

## Checklist
- [x] I have manually tested these changes
- [x] Post a link to the PR in the group chat

## Guidelines
- If your PR is not ready, mark it as a draft
- The `resolve conversation` button is for reviewers, not authors
  - (But add a 'done' comment or similar)

### CODE REVIEW

**Review by blakecduncan — APPROVED** [review]
I'm not the best with solidity but logically this makes sense 👍

**Comment by jzaki** [comment]
Who can update what is required on the client side for this PR to be merged? @jacque006 @kautukkundan @voltrevo

File: contracts/test/walletAction-test.ts [hunk 1]
```diff
@@ -307,21 +317,21 @@ describe("WalletActions", async function () {
...
 
+    // Single event "WalletOperationProcessed(address indexed wallet, uint256 nonce, bool success, bytes[] results)"
+    // Get the first (only) result from "results" argument.
+    const result = r.events[0].args.results[0]; // For errors this is "Error(string)"
+    const errorArgBytesString: string = "0x" + result.substring(10); // remove methodId (4bytes after 0x)
+    const errorString = ethers.utils.defaultAbiCoder.decode(
+      ["string"],
+      errorArgBytesString,
+    )[0]; // decoded bytes is a string of the action index that errored.
+    expect(errorString).to.equal("1 - ERC20: transfer from the zero address");
```
Comment (jacque006) [inline contracts/test/walletAction-test.ts:333]: :tada:

**Comment by jacque006** [comment]
@kautukkundan Will update

**Pull request merged**